test-suite/js/src/test/scala/org/scalajs/testsuite/compiler/RegressionTest.scala

```diff
@@ -440,7 +442,15 @@ class RegressionTest {
   @Test def null_asInstanceOf_Unit_should_succeed_issue_1691(): Unit = {
     def getNull(): Any = null
     val x = getNull().asInstanceOf[Unit]: Any
```

Comment:
Mostly to avoid scalac's special treatment of `<literal null>.asInstanceOf[X]`. It does have the benefit to test our constant-folder of that pattern, once `getNull()` is inlined; and of our run-time implementation, when the optimizer is disabled.